sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -29,9 +29,9 @@
 #define SOF_RT5682_MCLK_EN			BIT(3)
 #define SOF_RT5682_MCLK_24MHZ			BIT(4)
 #define SOF_SPEAKER_AMP_PRESENT		BIT(5)
-#define SOF_RT5682_SSP_AMP(quirk)		((quirk) & GENMASK(8, 6))
-#define SOF_RT5682_SSP_AMP_MASK			(GENMASK(8, 6))
 #define SOF_RT5682_SSP_AMP_SHIFT		6
+#define SOF_RT5682_SSP_AMP(quirk)		((quirk) << SOF_RT5682_SSP_AMP_SHIFT)
```

Comment:
this makes no sense. You are passing a 1 as a parameter, so this can be replaced by SOF_SPEAKER_AMP_PRESENT

You don't need these 3 definitions line 32..34 at all.

Reply:
actually the AMP_PRESENT only tells if AMP is present. The quirt is to tell the SSP its connected to.
We can have a platform where AMP is present and can be on SSP2 or maybe SSP0 as well.

Hence have separate definitions.

Reply:
ok, then you do need a mask to protect the quirk.
I should be 

define SOF_RT5682_SSP_AMP_MASK			(GENMASK(8, 6)
define SOF_RT5682_SSP_AMP(quirk)		(((quirk) << SOF_RT5682_SSP_AMP_SHIFT) && SOF_RT5682_SSP_AMP_MASK)

and you use it as 

ssp_amp = (sof_rt5682_quirk & SOF_RT5682_SSP_AMP_MASK) >>			s
			SOF_RT5682_SSP_AMP_SHIFT;

Reply:
This seems to be same as 1st way i did in v1 ? should i bring back that way as in v1 then

Reply:
The problem is that you use quirk as a parameter when we also have a sof_rt5682_quirk variable, so this is confusing. You are not shifting the quirk, you have shifting a configuration to fit it into the quirk.

Reply:
@pierre, i did the way you suggested in above comments and submitted the patch again.